File: analyze_questions.js
const fs = require('fs');

// Ler o arquivo de questões
const data = JSON.parse(fs.readFileSync('questions/Padrão.txt', 'utf-8'));

// Extrair matérias e dificuldades únicas
const materiasSet = new Set();
const dificuldadesSet = new Set();
const materiaCount = {};
const dificuldadeCount = {};

data.forEach(q => {
    const mat = q.materia || '';
    const dif = q.dificuldade || '';

    materiasSet.add(mat);
    dificuldadesSet.add(dif);

    materiaCount[mat] = (materiaCount[mat] || 0) + 1;
    dificuldadeCount[dif] = (dificuldadeCount[dif] || 0) + 1;
});

// Converter para arrays e ordenar
const materias = Array.from(materiasSet).sort();
const dificuldades = Array.from(dificuldadesSet).sort();

// Exibir resultados
console.log('='.repeat(60));
console.log('ANÁLISE DE MATÉRIAS');
console.log('='.repeat(60));
console.log(`\nTotal de questões: ${data.length}\n`);

console.log('MATÉRIAS ÚNICAS (ordenadas alfabeticamente):');
console.log('-'.repeat(60));
materias.forEach((m, i) => {
    console.log(`${(i + 1).toString().padStart(2)}. '${m}' → ${materiaCount[m]} questões`);
});
console.log(`\nTotal de matérias diferentes: ${materias.length}`);

console.log('\n' + '='.repeat(60));
console.log('ANÁLISE DE DIFICULDADES');
console.log('='.repeat(60));
console.log('\nDIFICULDADES ÚNICAS:');
console.log('-'.repeat(60));
dificuldades.forEach((d, i) => {
    console.log(`${i + 1}. '${d}' → ${dificuldadeCount[d]} questões`);
});
console.log(`\nTotal de níveis diferentes: ${dificuldades.length}`);

// Verificar inconsistências específicas
console.log('\n' + '='.repeat(60));
console.log('VERIFICAÇÃO DE INCONSISTÊNCIAS');
console.log('='.repeat(60));

// Verificar "Média" vs "Médio"
const hasMedia = dificuldadeCount['Média'] > 0;
const hasMedio = dificuldadeCount['Médio'] > 0;

if (hasMedia && hasMedio) {
    console.log(`\n⚠️ ENCONTRADO: Mistura de 'Média' (${dificuldadeCount['Média']}) e 'Médio' (${dificuldadeCount['Médio']})`);
} else if (hasMedia) {
    console.log(`\n✅ Todas usam 'Média' (${dificuldadeCount['Média']} questões)`);
} else if (hasMedio) {
    console.log(`\n✅ Todas usam 'Médio' (${dificuldadeCount['Médio']} questões)`);
}

// Verificar matérias compostas (com "/")
const materiasCompostas = materias.filter(m => m.includes('/'));
if (materiasCompostas.length > 0) {
    console.log(`\n⚠️ MATÉRIAS COMPOSTAS ENCONTRADAS (${materiasCompostas.length}):`);
    materiasCompostas.forEach(mat => {
        console.log(`   - '${mat}' → ${materiaCount[mat]} questões`);
    });
}

console.log('\n' + '='.repeat(60));

File: analyze_questions.py
import json
from collections import Counter

# Ler o arquivo de questões
with open('questions/Padrão.txt', 'r', encoding='utf-8') as f:
    questoes = json.load(f)

# Extrair matérias e dificuldades únicas
materias = [q.get('materia', '') for q in questoes]
dificuldades = [q.get('dificuldade', '') for q in questoes]

# Contar ocorrências
materias_counter = Counter(materias)
dificuldades_counter = Counter(dificuldades)

# Exibir resultados
print("=" * 60)
print("ANÁLISE DE MATÉRIAS")
print("=" * 60)
print(f"\nTotal de questões: {len(questoes)}\n")

print("MATÉRIAS ÚNICAS (ordenadas alfabeticamente):")
print("-" * 60)
for i, materia in enumerate(sorted(materias_counter.keys()), 1):
    count = materias_counter[materia]
    print(f"{i:2d}. '{materia}' → {count} questões")

print(f"\nTotal de matérias diferentes: {len(materias_counter)}")

print("\n" + "=" * 60)
print("ANÁLISE DE DIFICULDADES")
print("=" * 60)
print("\nDIFICULDADES ÚNICAS:")
print("-" * 60)
for i, dificuldade in enumerate(sorted(dificuldades_counter.keys()), 1):
    count = dificuldades_counter[dificuldade]
    print(f"{i}. '{dificuldade}' → {count} questões")

print(f"\nTotal de níveis diferentes: {len(dificuldades_counter)}")

# Verificar inconsistências específicas
print("\n" + "=" * 60)
print("VERIFICAÇÃO DE INCONSISTÊNCIAS")
print("=" * 60)

# Verificar "Média" vs "Médio"
if 'Média' in dificuldades_counter and 'Médio' in dificuldades_counter:
    print(f"\n⚠️ ENCONTRADO: Mistura de 'Média' ({dificuldades_counter['Média']}) e 'Médio' ({dificuldades_counter['Médio']})")
elif 'Média' in dificuldades_counter:
    print(f"\n✅ Todas usam 'Média' ({dificuldades_counter['Média']} questões)")
elif 'Médio' in dificuldades_counter:
    print(f"\n✅ Todas usam 'Médio' ({dificuldades_counter['Médio']} questões)")

# Verificar matérias compostas (com "/")
materias_compostas = [m for m in materias_counter.keys() if '/' in m]
if materias_compostas:
    print(f"\n⚠️ MATÉRIAS COMPOSTAS ENCONTRADAS ({len(materias_compostas)}):")
    for mat in sorted(materias_compostas):
        print(f"   - '{mat}' → {materias_counter[mat]} questões")

print("\n" + "=" * 60)
